### DISCUSSION TITLE
Some questions and asking for help

### DISCUSSION THREAD

**Discussion by mahozad** [bytedeco/javacv#1956] [Q&A]
Hi.

First of all, this library is a gem. Thanks for your effort.

### Project Panama
Is this library going to be migrated to [Project Panama](https://openjdk.org/projects/panama/)? (Related [YouTube video](https://www.youtube.com/watch?v=4xFV-A7JToY&ab_channel=Java)).
The [JavaCPP project](https://github.com/bytedeco/javacpp) is even mentioned in the [Java JEP](https://openjdk.org/jeps/434#:~:text=%2C%20JNR%20and-,JavaCPP,-.%20While%20these%20frameworks).
Some famous libraries like [Apache Lucene](https://github.com/apache/lucene) seem to be [starting to migrate to and use Project Panama](https://github.com/apache/lucene/pull/912).

### A video player for Compose on desktop
I'm trying to create a simple video player component for Desktop applications developed with **[Compose Multiplatform](https://github.com/JetBrains/compose-jb)** (don't know why its repository is called *compose-jb*). **Compose Multiplatform** is a port of [Google Jetpack Compose Android framework](https://developer.android.com/jetpack/compose) for Desktop/Web/IOS (sorry for this description if you are already familiar with Compose).

Currently, there is no video player component for this framework on desktop. The [authors/maintainers recommend](https://github.com/JetBrains/compose-jb/tree/master/experimental/components/VideoPlayer) using the [vlcj library](https://github.com/caprica/vlcj) which, as far as I know, [requires VLC to have been installed on the system](https://capricasoftware.co.uk/projects/vlcj-4/tutorials/prerequisites).

I have created this very basic and rudimentary video player. It does not have any buffering, cannot pause/resume etc. I'm surprised that it works at all:

https://github.com/mahozad/compose-video-player

I've tried to sync the video to the audio but for some videos (especially [.ts files](https://en.wikipedia.org/wiki/MPEG_transport_stream), including the one that the app plays by default) it does not work well enough and video/audio are out of sync.

 - Also, this is not very performant. If I make the width and height a little larger, then the player cannot be fast enough and keep up with the video frame rate. On that matter, could you please provide a converter to directly convert a `Frame` to Compose [`ImageBitmap`](https://github.com/androidx/androidx/blob/androidx-main/compose/ui/ui-graphics/src/commonMain/kotlin/androidx/compose/ui/graphics/ImageBitmap.kt)?
- For the sample.ts file (which is played when running the application), I had to multiply the audio sample rate by `2` so the audio plays at normal speed:

  https://github.com/mahozad/compose-video-player/blob/4c90111b2eb8e09aee2fe9e663c4b955be11651d/src/main/kotlin/VideoPlayer.kt#L102

  Why should I do that? It is because the video is interlaced?

**Comment by saudet** [comment]
Are you asking about making presets for libvlc?

**Comment by mahozad** [comment]
I'm trying to create [a very simple video player](https://github.com/mahozad/compose-video-player) with FFmpeg (*org.bytedeco:ffmpeg*) for [Compose Multiplatform](https://github.com/JetBrains/compose-jb).

I guess you didn't have time to read the whole text above.
Sorry if it is too long.

**Reply by saudet** [reply]
I understand that, but it sounds a lot easier to use an existing player like VLC then try to make one on your own. Did you try the one in JavaFX?

**Reply by saudet** [reply]
@johanvos Are you aware of effort to integrate JavaFX's video player to https://github.com/JetBrains/compose-jb?
